cloudinit/sources/helpers/azure.py

```diff
@@ -258,6 +258,51 @@ def _get_dhcp_endpoint_option_name():
     return azure_endpoint
 
 
+@azure_ds_telemetry_reporter
+def http_with_retries(url, **kwargs) -> str:
+    exc = None
+
+    max_readurl_attempts = 240
+    default_readurl_timeout = 5
+    periodic_logging_attempts = 12
+
+    if 'timeout' not in kwargs:
+        kwargs['timeout'] = default_readurl_timeout
+
+    # remove kwargs that cause url_helper.readurl to retry,
+    # since we are already implementing our own retry logic.
+    if 'retries' in kwargs:
+        kwargs.pop('retries', None)
+        LOG.warning(
+            'retries kwarg passed in for communication with Azure endpoint.')
```

Comment:
Since we aren't using the value of retries, no need to if and pop. Let's also just mention in logs what we are doing with that unexpected arg. "Ignoring..."

```suggestion
    if kwargs.pop('retries', None):
        LOG.warning(
            'Ignoring retries kwarg passed in for communication with Azure endpoint.')
```

Reply:
Done. https://github.com/canonical/cloud-init/pull/619/commits/d8211a3cf6a514996ec6e5f3f45f6862ef52960a